redis/src/sentinel.rs

```diff
@@ -0,0 +1,795 @@
+    /// then correctly responds to the SENTINEL MASTERS command, and then we check to
+    /// see if there is a master with the given name; if there is, we create a
+    /// client with that address (ip and port) and the extra information from the
+    /// `node_connection_info` parameter.
+    pub fn master_for(
+        &mut self,
+        service_name: &str,
+        node_connection_info: Option<&SentinelNodeConnectionInfo>,
+    ) -> RedisResult<Client> {
+        let connection_info =
+            self.find_master_address(service_name, node_connection_info.unwrap_or_default())?;
+        Client::open(connection_info)
+    }
+
+    /// Connects to the first available replica of the given master name.
+    ///
+    /// After listing the valid replicas (those without a sdown or odown flag) for the
+    /// given master name, we start trying to connect to then one by one, starting at a
+    /// random index (simple random derived from the nanoseconds part of the current
+    /// unix timestamp).
+    pub fn replica_for(
+        &mut self,
+        service_name: &str,
+        node_connection_info: Option<&SentinelNodeConnectionInfo>,
+    ) -> RedisResult<Client> {
+        let addresses = self
+            .find_valid_replica_addresses(service_name, node_connection_info.unwrap_or_default())?;
+        let start_index = random_replica_index(addresses.len());
+        try_connect_to_first_replica(&addresses, start_index)
+    }
+
+    /// Connects to the first available replica of the given master name, starting in a
+    /// different (incremented) index each time the function is called.
+    ///
+    /// After listing the valid replicas (those without a sdown or odown flag) for the
+    /// given master name, we start trying to connect to then one by one, starting at a
+    /// rotating index which gets incremented on each call to this function.
+    pub fn replica_rotate_for(
+        &mut self,
+        service_name: &str,
+        node_connection_info: Option<&SentinelNodeConnectionInfo>,
+    ) -> RedisResult<Client> {
+        let addresses = self
+            .find_valid_replica_addresses(service_name, node_connection_info.unwrap_or_default())?;
+        if !addresses.is_empty() {
+            self.replica_start_index = (self.replica_start_index + 1) % addresses.len();
+        }
+        try_connect_to_first_replica(&addresses, self.replica_start_index)
+    }
+}
+
+/// Enum defining the server types from a sentinel's point of view.
+#[derive(Debug, Clone)]
+pub enum SentinelServerType {
+    /// Master connections only
+    Master,
+    /// Replica connections only
+    Replica,
+}
+
+/// An alternative to the Client type which creates connections from clients created
+/// on-demand based on information fetched from the sentinels. Uses the Sentinel type
+/// internally. This is basic an utility to help make it easier to use sentinels but
+/// with an interface similar to the client (`get_connection` and
+/// `get_async_connection`).
```

Comment:
Just to clarify, please add to the doc that the client is limited to only returning one of replicas or masters, not both.

Reply:
Ok, done!